exporter/exporterhelper/request/request.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package request // import "go.opentelemetry.io/collector/exporter/exporterhelper/request"
+
+import (
+	"context"
+)
+
+// Request represents a single request that can be sent to an external endpoint.
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+type Request interface {
+	// Export exports the request to an external endpoint.
+	Export(ctx context.Context) error
+	// ItemsCount returns a number of basic items in the request where item is the smallest piece of data that can be
+	// sent. For example, for OTLP exporter, this value represents the number of spans,
+	// metric data points or log records.
+	ItemsCount() int
```

Comment:
One thing that you need to make sure, especially for batching is that it is very bad for some systems to split the metric across requests. Even if the limit may be at data points, you may want to create an issue for your batcher work.

Reply:
Makes sense. We can probably add comments to the batcher config recommending users to keep the difference between `MinSizeItems` and `MaxSizeItems` high and implement batching for OTLP exporter to make the best effort not to split metrics with those restrictions. WDYT?